client/src/Features/AssignmentLinks/useLink.hook.ts

```diff
@@ -5,16 +5,24 @@
 
 import { AssignmentLink } from '../../Models/AssignmentLink.model';
 import { useState, useEffect } from 'react';
+import { isValidURL, ValidateDescLength, ValidateTitleLength } from './AssignmentLinkEditor';
 
 type useLinkReturnType = [AssignmentLink, (link: AssignmentLink) => void, boolean];
 
 export const useLink = (assignmentLink: AssignmentLink): useLinkReturnType => {
   const [link, setLink] = useState<AssignmentLink>(assignmentLink);
-  const [canSave, setCanSave] = useState(!!link.url);
+
+  const titleIsValid = ValidateTitleLength(link.displayText) === "";

```

Comment:
The unfortunate part of handling it this way is that the implementation calls all 3 functions before deciding if the link is valid or not. In previous case, that was not required if any of the checks failed.

> Essentially, I liked the previous approach :).

Reply:
I have made the changes, that should help here!